classes/fields.php

```diff
@@ -99,543 +99,699 @@ public static function not_support( $type, $not_support ){
+			'section_break'      => self::get_section_break_field(),
+
+			//select
+			'dropdown'           => self::get_dropdown_field(),
+			'checkbox'           => self::get_checkbox_field(),
+			'radio'              => self::get_radio_field(),
+			'filtered_select2'   => self::get_filtered_select2_field(),
+			'date_picker'        => self::get_date_picker_field(),
+			'toggle_switch'      => self::get_toggle_switch_field(),
+			'color_picker'       => self::get_colorpicker_field(),
+			'states'             => self::get_states_field(),
+
+			//discontinued
+			'recaptcha'          => self::get_recaptcha_field(),
+		);
+
+		return $internal_fields;
+	}
+
+	public static function get_text_field()
+	{
+		return array(
+			"field"       => __( 'Single Line Text', 'caldera-forms' ),
+			"description" => __( 'Single Line Text', 'caldera-forms' ),
+			"file"        => CFCORE_PATH . "fields/generic-input.php",
+			"category"    => __( 'Basic', 'caldera-forms' ),
+			"setup"       => array(
+				"template" => CFCORE_PATH . "fields/text/config.php",
+				"preview"  => CFCORE_PATH . "fields/text/preview.php"
 			),
-			'hidden'           => array(
-				"field"       => __( 'Hidden', 'caldera-forms' ),
-				"description" => __( 'Hidden', 'caldera-forms' ),
-				"file"        => CFCORE_PATH . "fields/hidden/field.php",
-				"category"    => __( 'Basic', 'caldera-forms' ),
-				"static"      => true,
-				"setup"       => array(
-					"preview"       => CFCORE_PATH . "fields/hidden/preview.php",
-					"template"      => CFCORE_PATH . "fields/hidden/setup.php",
-					"not_supported" => array(
-						'hide_label',
-						'caption',
-						'required',
-					)
-				)
-			),
-			'email'            => array(
-				"field"       => __( 'Email Address', 'caldera-forms' ),
-				"description" => __( 'Email Address', 'caldera-forms' ),
-				'icon'          => CFCORE_URL . 'assets/build/images/envelope-o.svg',
-				"file"        => CFCORE_PATH . "fields/generic-input.php",
-				"category"    => __( 'Basic', 'caldera-forms' ),
-				"setup"       => array(
-					"preview"  => CFCORE_PATH . "fields/email/preview.php",
-					"template" => CFCORE_PATH . "fields/email/config.php"
-				)
-			),
-			'button'           => array(
-				"field"       => __( 'Button', 'caldera-forms' ),
-				"description" => __( 'Button, Submit and Reset types', 'caldera-forms' ),
-				"file"        => CFCORE_PATH . "fields/button/field.php",
-				"category"    => __( 'Basic', 'caldera-forms' ),
-				"capture"     => false,
-				"setup"       => array(
-					"template"      => CFCORE_PATH . "fields/button/config_template.php",
-					"preview"       => CFCORE_PATH . "fields/button/preview.php",
-					"default"       => array(
-						'class' => 'btn btn-default',
-						'type'  => 'submit'
-					),
-					"not_supported" => array(
-						'hide_label',
-						'caption',
-						'required',
-						'entry_list'
-					)
+		);
+	}
+
+	public static function get_hidden_field()
+	{
+		return array(
```

Comment:
arrays should use bracket syntax, but don't worry #2362 will fix that for us. I know this is copying legacy code, don't worry about it.